server/broadcast.go

```diff
@@ -682,9 +682,11 @@ func (bsm *BroadcastSessionsManager) collectResults(submitResultsCh chan *Submit
 // the caller needs to ensure bsm.sessLock is acquired before calling this.
 func (bsm *BroadcastSessionsManager) completeSessionUnsafe(ctx context.Context, sess *BroadcastSession, tearDown bool) {
 	if tearDown {
-		if err := EndTranscodingSession(ctx, sess); err != nil {
-			clog.Errorf(ctx, "Error completing transcoding session: %q", err)
-		}
+		go func() {
```

Comment:
@yondonfu @iameli Are you guys happy with this? I've been looking at the B code downstream of here as well as the function that ends up being called on the O side and can't see a need for blocking.

Reply:
AFAICT you seem to be right that this call doesn't need to block for B.